src/stores/ExploreDataStore.ts

```diff
@@ -0,0 +1,130 @@
+import { makeAutoObservable } from "mobx";
+import { ConstraintInstance, TypeFormatTuple, WorkflowConfig, WorkflowSolution } from "./WorkflowTypes";
+import { makePersistable } from "mobx-persist-store";
+
+const emptyWorkflowConfig = () => {
+  return {
+    domain: undefined,
+    inputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple ],
+    outputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple],
+    constraints: [ {constraint: {id:"",label:""}} as ConstraintInstance ],
+    minSteps: 3,
+    maxSteps: 4,
+    timeout: 120,
+    solutionCount: 10,
+    run_id: ""
+  }
+}
+
+export class ExploreDataStore {
+
+  workflowConfig: WorkflowConfig = emptyWorkflowConfig();
+  workflowSolutions: WorkflowSolution[] = [];
+  isGenerating: boolean = false;
+  generationError: string = "";
+
+  constructor() {
+    makeAutoObservable(this, {}, { deep: true });
+    makePersistable(this, { 
+      name: "ExploreDataStore",
+      properties: ["workflowConfig", "workflowSolutions"],
+      storage: window.localStorage
+    });
+  }
+
+  inputsOutputsToJSON(values: TypeFormatTuple[], dataRoot: string, formatRoot: string) {
+    return values.filter(value => value[0] !== undefined && value[1] !== undefined && value[0]!.id !== "" && value[1]!.id !== "")
+      .map((value) => {
+        return {
+          [dataRoot]: [value[0]!.id.replace("http://edamontology.org/", "")],
```

Comment:
The configuration works with full IRIs as well. The short syntax is just there to allow easier manual configuration. Also, the given prefix might not be valid, in case an altered (or different) ontology is used. That's why the ontology_prexif field is part of the APE configuration.

Reply:
Good to know; not sure why I explicitly replace it then.

Reply:
The examples usually contain the shorter version, so probably that's the reason.